applications/crossbar/src/cb_context.erl

```diff
@@ -530,21 +529,22 @@ failed_error({'data_invalid'
 
     add_validation_error(FailedKeyPath
                          ,<<"maxLength">>
-                         ,build_error_message(cb_context:api_version(Context)
-                                              ,<<"String must not be more than ", MaxLen/binary, " characters">>
-                                              ,Maximum
-                                             )
+                         ,[{<<"message">>, <<"String must not be more than ", MaxLen/binary, " characters">>}
+                           ,{<<"target">>, Maximum}
+                         ]
                          ,Context
                         );
 failed_error({'data_invalid'
-              ,_FailedSchemaJObj
+              ,FailedSchemaJObj
               ,'not_in_enum'
               ,_FailedValue
               ,FailedKeyPath
              }, Context) ->
     add_validation_error(FailedKeyPath
                          ,<<"enum">>
-                         ,<<"Value not found in enumerated list of values">>
+                         ,[{<<"message">>, <<"Value not found in enumerated list of values">>}
```

Comment:
As we discussed, I think all calls to `add_validation_error/4` should pass a json object in the third argument.